pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -49,18 +50,26 @@ type DiffListChange struct {
 	Diff *diff.Result
 }
 
-func Diff(old, new Manifest, opts ...diff.Option) (*diff.Result, error) {
+func Diff(old, new Manifest, logger *zap.Logger, opts ...diff.Option) (*diff.Result, error) {
 	if old.Key.IsSecret() && new.Key.IsSecret() {
 		var err error
 		new.u, err = normalizeNewSecret(old.u, new.u)
 		if err != nil {
 			return nil, err
 		}
 	}
+
+	normalized, err := remarshal(old.u)
+	if err != nil {
+		logger.Info(fmt.Sprintf("Some error occurred while remarshaling %s", err.Error()))
+	} else {
+		old.u = normalized
+	}
+
 	return diff.DiffUnstructureds(*old.u, *new.u, opts...)
```

Comment:
nit: To avoid updating the passed `old` parameter.

```suggestion
	normalized, err := remarshal(old.u)
	if err != nil {
		logger.Info(fmt.Sprintf("Some error occurred while remarshaling %s", err.Error()))
		return diff.DiffUnstructureds(*old.u, *new.u, opts...)
	}

	return diff.DiffUnstructureds(*normalized, *new.u, opts...)
```